aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/TypeConverter.java

```diff
@@ -81,6 +83,26 @@ static JavaFieldType getJavaFieldType(@NonNull ModelField field) {
         }
     }
 
+    /**
+     * Gets the {@link JavaFieldType} from the value class.
+     * @param value The value to guess the type from.
+     * @return the {@link JavaFieldType} from the value class.
+     */
+    public static JavaFieldType getJavaFieldTypeFromValue(@NonNull Object value) {
```

Comment:
I'm not sure how this would be a benefit, since the value we have at the time we call this is of type `Object`. We're reading it using reflection from a customer provided type, so there's no way to know the actual type beforehand in a way the `T` would be relevant.

Thoughts?